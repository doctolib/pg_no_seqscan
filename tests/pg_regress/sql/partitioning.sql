-- Test partitioning support
LOAD 'pg_no_seqscan';
SET pg_no_seqscan.level = ERROR;

CREATE TABLE partitioned_foo (id bigint) PARTITION BY RANGE (id);
CREATE TABLE partitioned_foo_1 PARTITION OF partitioned_foo FOR VALUES FROM (1) TO (5);
CREATE TABLE partitioned_foo_2 PARTITION OF partitioned_foo FOR VALUES FROM (5) TO (11);

SET pg_no_seqscan.check_tables = 'partitioned_foo';

-- Querying parent should error
EXPLAIN (COSTS OFF) SELECT * FROM partitioned_foo;
SELECT * FROM partitioned_foo;

-- cleanup
Drop table partitioned_foo;