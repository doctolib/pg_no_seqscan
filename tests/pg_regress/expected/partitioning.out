-- Test partitioning support
SET pg_no_seqscan.level = ERROR;
CREATE TABLE partitioned_foo (id bigint) PARTITION BY RANGE (id);
CREATE TABLE partitioned_foo_1 PARTITION OF partitioned_foo FOR VALUES FROM (1) TO (5);
CREATE TABLE partitioned_foo_2 PARTITION OF partitioned_foo FOR VALUES FROM (5) TO (11);
SET pg_no_seqscan.check_tables = 'partitioned_foo';
-- Querying parent should error
EXPLAIN SELECT * FROM partitioned_foo;
                                QUERY PLAN                                 
---------------------------------------------------------------------------
 Append  (cost=0.00..87.80 rows=4520 width=8)
   ->  Seq Scan on partitioned_foo_1  (cost=0.00..32.60 rows=2260 width=8)
   ->  Seq Scan on partitioned_foo_2  (cost=0.00..32.60 rows=2260 width=8)
(3 rows)

SELECT * FROM partitioned_foo;
ERROR:  A 'Sequential Scan' on partitioned_foo has been detected.
  - Run an EXPLAIN on your query to check the query plan.
  - Make sure the query is compatible with the existing indexes.

Query: SELECT * FROM partitioned_foo;

-- cleanup
Drop table partitioned_foo;
